packages/opentelemetry-resources/test/detectors/node/EnvDetector.test.ts

```diff
@@ -26,7 +26,7 @@ describeNode('envDetector() on Node.js', () => {
   describe('with valid env', () => {
     before(() => {
       process.env.OTEL_RESOURCE_ATTRIBUTES =
-        'k8s.pod.name="pod-xyz-123",k8s.cluster.name="c1",k8s.namespace.name="default"';
+        'k8s.pod.name="pod xyz 123",k8s.cluster.name="c1",k8s.namespace.name="default"';
```

Comment:
can we keep the value with the dashes and add a new one? it is the only one with special characters.

Reply:
Would you mind if I improved this test bed? The `assertK8sResource` assertion is not asserting the keys match the values because the `validations` keys (`clusterName`, `namespaceName`, `podName`, `deploymentName`) are all `undefined`

Reply:
I definitely would not mind

Reply:
Tried my best to keep the changes in the test minimal. Looks like `EnvDetector` never correctly trimmed the leading and trailing double quotes before and it wasn't caught by the tests because the `assertK8sResource` was not used as intended.